### PR TITLE
TF-3226 Auto complete should rely minimum length to auto completable on the capability object

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#3226

### CODE REVIEW

File: contact/lib/contact/model/autocomplete_capability.dart
Comment (tddang-linagora) [inline contact/lib/contact/model/autocomplete_capability.dart:1]: Check github action warning

File: contact/lib/contact/model/autocomplete_capability.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+part 'autocomplete_capability.g.dart';
+
+@JsonSerializable(
+  explicitToJson: true,
+  includeIfNull: false,
+  converters: [
+    UnsignedIntNullableConverter()
+  ]
+)
+class AutocompleteCapability extends CapabilityProperties {
```
Comment (tddang-linagora) [inline contact/lib/contact/model/autocomplete_capability.dart:15]: Should this be defined in `jmap-dart-client` instead?

Comment (dab246) [reply contact/lib/contact/model/autocomplete_capability.dart:15]: No, it is an extension so we have already created a module to handle it.

File: contact/lib/contact/model/autocomplete_capability.dart
Comment (dab246) [reply contact/lib/contact/model/autocomplete_capability.dart:1]: No problem, only name is not spell

**Review by tddang-linagora — DISMISSED** [review]

File: lib/features/base/base_controller.dart [hunk 2]
```diff
@@ -471,4 +472,12 @@ abstract class BaseController extends GetxController
...
     await cachingManager.closeHive();
   }
+
+  int getMinInputLengthAutocomplete({
+    required Session session,
+    required AccountId accountId,
+  }) {
+    final minInputLength = session.getMinInputLengthAutocomplete(accountId);
+    return minInputLength?.value.toInt() ?? AppConfig.defaultMinInputLengthAutocomplete;
+  }
```
Comment (hoangdat) [inline lib/features/base/base_controller.dart:482]: Why need to put it in controller? How about to use extension or util class?

Comment (dab246) [reply lib/features/base/base_controller.dart:482]: We already had handle `getMinInputLengthAutocomplete` in session extension and wirte unit test for it. This function to get value default `minInputLengthAutocomplete` when minInputLength from session is null. Additional, we only use it on all controller.

File: test/features/base/base_controller_test.dart [hunk 3]
```diff
@@ -201,4 +211,85 @@ void main() {
...
   });
+
+  group('BaseController::getMinInputLengthAutocomplete::test', () {
+    late MockBaseController mockBaseController;
+
+    setUp(() {
+      mockBaseController = MockBaseController();
+    });
+
+    test('SHOULD return session min input length WHEN AutocompleteCapability available', () {
```
Comment (hoangdat) [inline test/features/base/base_controller_test.dart:222]: should one more test case:
```
test('SHOULD return default min input length WHEN AutocompleteCapability available, but no minInputLenght', () {
```

Comment (dab246) [reply test/features/base/base_controller_test.dart:222]: done

**Review by hoangdat — APPROVED** [review]

**Review by tddang-linagora — APPROVED** [review]

**Pull request merged**